File: leibniz/perl/leibniz.pl
use strict;
use warnings;
use utf8;

sub powersign {
    my ($n) = @_;
    if ($n % 2 == 0) {
        return 1;
    } else {
        return -1;
    }
}

sub leibniz {
    my ($n) = @_;
    my $s = 0;
    for (my $i = 0; $i <= $n; $i++) {
        $s += powersign($i) / (2 * $i + 1)
    }
    return 4 * $s;
}

print leibniz(@ARGV);
print "\n";

File: leibniz/ocaml/leibniz.ml
let powersign n =
  if n mod 2 = 0 then 1 else -1

let leibniz n =
  let s = ref 0. in
  for k = 0 to n do
    s := !s +. float (powersign k) /. float (2 * k + 1)
  done ;
  4. *. !s

let n = int_of_string Sys.argv.(1) ;;
Printf.printf "%.16f\n" (leibniz n)
